voltdbclient/client.go

```diff
@@ -0,0 +1,192 @@
+/* This file is part of VoltDB.
+ * Copyright (C) 2008-2016 VoltDB Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with VoltDB.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package voltdbclient
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+)
+
+// Client is a single connection to a single node of a VoltDB database
+type Client struct {
+	tcpConn  *net.TCPConn
+	connData *connectionData
+}
+
+// connectionData are the values returned by a successful login.
+type connectionData struct {
+	hostId      int32
+	connId      int64
+	leaderAddr  int32
+	buildString string
+}
+
+// NewClient creates an initialized, authenticated Client.
+func NewClient(user string, passwd string, hostAndPort string) (*Client, error) {
+	var client = new(Client)
+	var err error
+	var raddr *net.TCPAddr
+	var login bytes.Buffer
+
+	if raddr, err = net.ResolveTCPAddr("tcp", hostAndPort); err != nil {
```

Comment:
do you need to implement a reverse dns cache as we did in Java if a use net is misconfigured it could take a long time to connect